### DISCUSSION TITLE
How can I replace the native keyboard dictation button with an online speech-to-text service, e.g: Azure?

### DISCUSSION THREAD

**Discussion by jagandecapri** [kirillzyusko/react-native-keyboard-controller#631] [Q&A] [answered]
Hi,

Is it possible to the following logic with react-native-keyboard-controller?

- When the Azure Speech-to-Text service is reachable within X second(s), then the dictation button will use the service when tapped by the user
- Else, it uses the device's (iOS/Android) dictation

The button that I'm referring to is as following:

![IMG_6716C7F83ADA-1](https://github.com/user-attachments/assets/6fa8f133-01d4-46af-9ff3-0e253c7d6a08)

Thank you!

**Comment by kirillzyusko** [comment, marked as answer]
@jagandecapri This is a system button and you (as a developer) don't have an access to customize its position/press handlers etc.

Potentially you may draw a view above the keyboard with your custom buttons using [KeyboardStickyView](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/components/keyboard-sticky-view) and place your custom icons there and then use `azure`/`react-native-voice`/etc. (or alternatively draw custom mic icon that will follow [carret](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/hooks/input/use-focused-input-handler#onselectionchange) position)

Did you see any apps that use custom voice recognition module (i. e. azure) if you press system mic button?

P. S. I searched more and I think the best what I can do is to emit an event `inputMethodChanged` with a payload (i. e. type=`english`/`voice` etc.) But I don't think you'll be able to start recording of audio, since most likely it'll interrupt voice recognition from a keyboard.

**Reply by jagandecapri** [reply]
Hi @kirillzyusko ,

Thank you for the detailed reply. I have not seen this behaviour in any app. I was just wondering whether it would be possible. I'll explore other UX options.

**Reply by kirillzyusko** [reply]
@jagandecapri I think it's near to impossible, but if you find any app, then please share it here 🙏

**Reply by jagandecapri** [reply]
@kirillzyusko Noted. Thank you!
